conn/i2c/i2c.go

```diff
@@ -109,6 +110,20 @@ func (d *Dev) Duplex() conn.Duplex {
 	return conn.Half
 }
 
-//
+// Addr implements flag.Value.
+type Addr uint16
+
+// Set sets the Addr to a value respresented by the string s. Values maybe in
+// decimal or hexadecimal form.
+func (a *Addr) Set(s string) error {
+	u, err := strconv.ParseUint(s, 0, 16)
+	if err != nil {
+		return errI2CSetError
+	}
+	*a = Addr(u)
```

Comment:
Valid addresses are on 11 bits (when including R/W bit); so values above 0x7ff are invalid.
https://www.totalphase.com/support/articles/200349176-7-bit-8-bit-and-10-bit-I2C-Slave-Addressing gives a nice explanation.